src/libs/kata-types/src/config/mod.rs

```diff
@@ -0,0 +1,303 @@
+pub fn validate_path_pattern<P: AsRef<Path>>(patterns: &[String], path: P) -> Result<()> {
+    let path = path
+        .as_ref()
+        .to_str()
+        .ok_or_else(|| eother!("Invalid path {}", path.as_ref().to_string_lossy()))?;
+    for p in patterns.iter() {
+        if let Ok(glob) = glob::Pattern::new(p) {
+            if glob.matches(path) {
+                return Ok(());
+            }
+        }
+    }
+
+    Err(eother!("Path {} is not permitted", path))
+}
+
+/// Kata configuration information.
+pub struct KataConfig {
+    config: TomlConfig,
+    agent: String,
+    hypervisor: String,
+}
+
+impl KataConfig {
+    /// Set the default Kata configuration object.
+    ///
+    /// The default Kata configuration information is loaded from system configuration file.
+    pub fn set_default_config(config: TomlConfig, hypervisor: &str, agent: &str) {
+        let kata = KataConfig {
+            config,
+            agent: agent.to_string(),
+            hypervisor: hypervisor.to_string(),
+        };
+        *KATA_ACTIVE_CONFIG.lock().unwrap() = Arc::new(kata);
+    }
+
+    /// Get the default Kata configuration object.
+    ///
+    /// The default Kata configuration information is loaded from system configuration file.
+    pub fn get_default_config() -> Arc<KataConfig> {
+        KATA_ACTIVE_CONFIG.lock().unwrap().clone()
+    }
+
+    /// Set the active Kata configuration object.
+    ///
+    /// The active Kata configuration information is default configuration information patched
+    /// with tunable configuration information from annotations.
+    pub fn set_active_config(config: TomlConfig, hypervisor: &str, agent: &str) {
+        let kata = KataConfig {
+            config,
+            agent: agent.to_string(),
+            hypervisor: hypervisor.to_string(),
+        };
+        *KATA_ACTIVE_CONFIG.lock().unwrap() = Arc::new(kata);
+    }
+
+    /// Get the active Kata configuration object.
+    ///
+    /// The active Kata configuration information is default configuration information patched
+    /// with tunable configuration information from annotations.
+    pub fn get_active_config() -> Arc<KataConfig> {
+        KATA_ACTIVE_CONFIG.lock().unwrap().clone()
+    }
+    /// Get the config in use
+    pub fn get_config(&self) -> &TomlConfig {
+        &self.config
+    }
+
+    /// Get the agent configuration in use.
+    pub fn get_agent(&self) -> Option<&Agent> {
+        if !self.agent.is_empty() {
+            self.config.agent.get(&self.agent)
+        } else {
+            None
+        }
+    }
+
+    /// Get the hypervisor configuration in use.
+    pub fn get_hypervisor(&self) -> Option<&Hypervisor> {
+        if !self.hypervisor.is_empty() {
+            self.config.hypervisor.get(&self.hypervisor)
+        } else {
+            None
+        }
+    }
+}
+
+lazy_static! {
+    static ref KATA_DEFAULT_CONFIG: Mutex<Arc<KataConfig>> = {
+        let config = TomlConfig::load("").unwrap();
```

Comment:
If loading from empty string is only used to identity that the config is not initialized yet, it's better to use an `Option<TomlConfig>` type.

Reply:
it will be fixed in next version